src/liboslexec/closure.cpp

```diff
@@ -76,7 +76,7 @@ print_component_value(std::ostream &out, ShadingSystemImpl *ss,
     else if (type == TypeDesc::TypeVector)
         out << "(" << ((Vec3 *)data)->x << ", " << ((Vec3 *)data)->y << ", " << ((Vec3 *)data)->z << ")";
     else if (type == TypeDesc::TypeString)
-        out << "\"" << ((ustring *)data)->c_str() << "\"";
+        out << "\"" << ((ustring *)data)->string() << "\"";
```

Comment:
Ah, good catch. Though it occurs to me that you could also use

  ...  << *((ustring *)data) << ...

ustring knows how to write itself to a stream (safely!) without needing ->string() or ->c_str().